### PR TITLE
[Rest] Set null the redirect if empty on PageController

### PR DESCRIPTION
**Pull request by fkroockmann**
Currently redirect is always set as empty

### CODE REVIEW

File: Rest/Controller/PageController.php [hunk 1]
```diff
@@ -213,11 +213,14 @@ public function postAction(Layout $layout, Request $request, Page $parent = null
             $builder->setSite($this->getApplication()->getSite());
         }
 
+        $requestRedirect = $request->request->get('redirect');
```
Comment (mickaelandrieu) [inline Rest/Controller/PageController.php]:
```php
$requestRedirect = $request->request->get('redirect', null);
```

File: Rest/Controller/PageController.php [hunk 2]
```diff
@@ -295,11 +298,15 @@ public function putAction(Page $page, Layout $layout, Request $request)
         $page->setLayout($layout);
         $this->trySetPageWorkflowState($page, $this->getEntityFromAttributes('workflow'));
 
+
+        $requestRedirect = $request->request->get('redirect');
+        $redirect = ($requestRedirect === '' || $requestRedirect === null) ? null : $requestRedirect;
```
Comment (mickaelandrieu) [inline Rest/Controller/PageController.php]:
```php
$requestRedirect = $request->request->get('redirect', null);
```

File: Rest/Controller/PageController.php (hunk 1 above)
Comment (fkroockmann) [inline Rest/Controller/PageController.php]: This test will tell if the parameter does not exist, you set a value to null. Except that the current problem is that the "redirect" is knowledgeable but it is an empty string and an empty string is not a url ...

File: Rest/Controller/PageController.php (hunk 2 above)
Comment (fkroockmann) [inline Rest/Controller/PageController.php]: This test will tell if the parameter does not exist, you set a value to null. Except that the current problem is that the "redirect" is knowledgeable but it is an empty string and an empty string is not a url ...

Comment (mickaelandrieu) [inline Rest/Controller/PageController.php]: ok, i merge

File: Rest/Controller/PageController.php (hunk 1 above)
Comment (eric-chau) [inline Rest/Controller/PageController.php]: Anyway ``$requestRedirect = $request->request->get('redirect', null);`` is the default behavior, check the api.

Comment (fkroockmann) [inline Rest/Controller/PageController.php]: What api ?

Comment (fkroockmann) [inline Rest/Controller/PageController.php]: Yes, it's a js problem

Comment (eric-chau) [inline Rest/Controller/PageController.php]: ``Symfony\Component\HttpFoundation\ParameterBag::get``. It returns ``null`` only when the key does not exist (here in post data), but I guess you submit it in your form (there is an ``input[type="text"]`` named redirect) and empty is translated to empty string, not null which is logical.

Comment (fkroockmann) [inline Rest/Controller/PageController.php]: Yes :), that's what I meant to mickae.
But the problem is not in php, i will revert this PR.

Comment (eric-chau) [inline Rest/Controller/PageController.php]: @fkroockmann, your PR still legit, we can also check it in PHP side. But maybe move the test into ``BackBee\NestedNode\Page::setRedirect`` by something like this:

```php
    public function setRedirect($redirect = null)
    {
        $redirect = trim($redirect);
        $this->_redirect = empty($redirect) ? null : $redirect;
        return $this;
    }
```

**Pull request merged**